src/db/result_set.cr

```diff
@@ -99,6 +104,16 @@ module DB
       {% end %}
     end
 
+    private def internal_read(**types : **T) forall T
+      {% begin %}
+        NamedTuple.new(
+          {% for name, type in T %}
+            {{ name }}: read({{type.instance}}),
```

Comment:
I see the point, however the only way I see to change that is to create tuple of named tuple or create a new structure